### PR TITLE
implement flavor specific service policies

### LINKED ISSUE #1261: implement flavor specific service policies
Currently service policies are at service level. This must be extend to allow CSPs to have common policies at service level and additional policies at flavor level. 

- [ ] Extend the data model to provide an optional "flavor_name" field for service policies. 
- [ ] if flavor name provided during policy creation, validate if the flavor name is valid, if not return the existing `FlavorInvalidException` exception. 
- [ ] If flavor name is not provided, then the policy is always executed during service deployment. 
- [ ] If flavor name is provided, then this policy is validated only if the service was ordered for that specific flavor.

### PR DESCRIPTION
**Pull request by baixinsui**
fixes #1261

### CODE REVIEW

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/PolicyValidator.java [hunk 1]
```diff
@@ -88,35 +88,46 @@ public void validateDeploymentWithPolicies(DeployTask deployTask) {
...
+
+            List<String> flavorPolicyList = servicePolicies.stream().filter(servicePolicy ->
+                            StringUtils.equals(flavorName, servicePolicy.getFlavorName())
+                                    && StringUtils.isNotBlank(flavorName))
                     .map(ServicePolicy::getPolicy).toList();
+
+            if (!CollectionUtils.isEmpty(flavorPolicyList)) {
+                log.info("Evaluate deployment plan with service policies for flavor: {}",
+                        flavorName);
+                servicePolicyList = flavorPolicyList;
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/PolicyValidator.java:116]: if policies without flavor and  policies with flavor exist, then both the policies are executed. 
Sorry, I think the requirement was not clear in the user story. 
We must execute the policies without a flavor always.

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/policy/servicepolicy/ServicePolicy.java [hunk 2]
```diff
@@ -45,6 +45,9 @@ public class ServicePolicy {
     @Schema(description = "The id of registered service template which the policy belongs to.")
     private UUID serviceTemplateId;
 
+    @Schema(description = "The flavor name which the policy belongs to.")
```
Comment (swaroopar) [inline modules/models/src/main/java/org/eclipse/xpanse/modules/models/policy/servicepolicy/ServicePolicy.java:48]: please extend the description here to -

The flavor name which the policy belongs to. If a flavor is not provided, then the policy will be executed for during service deployment of all flavors.

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/policy/servicepolicy/ServicePolicyCreateRequest.java [hunk 3]
```diff
@@ -26,6 +26,10 @@ public class ServicePolicyCreateRequest {
     @Schema(description = "The id of registered service template which the policy belongs to.")
     private UUID serviceTemplateId;
 
+
+    @Schema(description = "The flavor name which the policy belongs to.")
```
Comment (swaroopar) [inline modules/models/src/main/java/org/eclipse/xpanse/modules/models/policy/servicepolicy/ServicePolicyCreateRequest.java:30]: please extend the description here to -

The flavor name which the policy belongs to. If a flavor is not provided, then the policy will be executed for during service deployment of all flavors.

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/policy/servicepolicy/ServicePolicy.java (hunk 2 above)
Comment (baixinsui) [reply modules/models/src/main/java/org/eclipse/xpanse/modules/models/policy/servicepolicy/ServicePolicy.java]: Updated.

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/policy/servicepolicy/ServicePolicyCreateRequest.java (hunk 3 above)
Comment (baixinsui) [reply modules/models/src/main/java/org/eclipse/xpanse/modules/models/policy/servicepolicy/ServicePolicyCreateRequest.java]: Updated.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/PolicyValidator.java (hunk 1 above)
Comment (baixinsui) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/PolicyValidator.java]: Updated.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**